scrapy/czlscrape/spiders/cultura.py

```diff
@@ -0,0 +1,217 @@
+:?                                          # optional colon
+\s?                                         # optional whitespace
+(?:la numarul|la numar|la nr\.|la nr)?      # optional prefix
+\s?                                         # optional whitespace
+(\S*)                                       # fax number
+.*                                          # anywhere in the string
+'''
+
+FEEDBACK_CONTACT_EMAIL_PATTERN = r'''
+.*                                          # can start anywhere
+e-mail:?                                    # the relevant prefix
+\s?                                         # optional whitespace
+(\S*)                                       # email address
+.*                                          # anywhere in the string
+'''
+
+FEEDBACK_CONTACT_RULES = [
+    ('fax', FEEDBACK_CONTACT_FAX_PATTERN),
+    ('email', FEEDBACK_CONTACT_EMAIL_PATTERN),
+]
+
+ROMANIAN_MONTHS = {
+    'ianuarie': 1,
+    'februarie': 2,
+    'martie': 3,
+    'aprilie': 4,
+    'mai': 5,
+    'iunie': 6,
+    'iulie': 7,
+    'august': 8,
+    'septembrie': 9,
+    'octombrie': 10,
+    'noiembrie': 11,
+    'decembrie': 12
+}
+
+INITIATIVE_TYPE_RULES = [
+    (r'\b(lege)\b', 'LEGE'),
+    (r'\b(hotarare|hg)\b', 'HG'),
+    (r'\b(ordonanta de guvern|og)\b', 'OG'),
+    (r'\b(ordonanta de urgenta|oug)\b', 'OUG'),
+    (r'\b(ordin de ministru|ordin|ordinul|om)\b', 'OM'),
+]
+
+INITIATIVE_TEXT_CLEANUP_PATTERNS = [r'(Fisiere|Fișiere|Anexe)[^:]*:']
+
+INITIATIVE_TEXT_CLEANUP_HTML_TAGS = ['ul', 'a']
+
+DIACRITICS_RULES = [
+    (r'[șş]', 's'),
+    (r'[ȘŞ]', 'S'),
+    (r'[țţ]', 't'),
+    (r'[ȚŢ]', 'T'),
+    (r'[ăâ]', 'a'),
+    (r'[ĂÂ]', 'A'),
+    (r'[î]', 'i'),
+    (r'[Î]', 'I'),
+]
+
+
+def remove_html_tags(selector: Selector, tags: list) -> Selector:
+    dom_element = selector.root
+    for tag in tags:
+        html_tag_selector = selector.xpath(tag)
+        if html_tag_selector:
+            dom_element.remove(html_tag_selector[0].root)
+    return selector
+
+
+def extract_text(selector: Selector) -> str:
+    return selector.xpath('string(.)').extract_first().strip()
+
+
+def collapse_blank_lines(text: str) -> str:
+    lines = re.sub("\n\s*\n{2,}", "\n\n", text).split("\n\n")
+    return "\n\n".join(line.strip() for line in lines)
+
+
+def remove_diacritics(text: str) -> str:
+    result = text
+    for search_pattern, replacement in DIACRITICS_RULES:
+        result = re.sub(search_pattern, replacement, result)
+    return result
+
+
+def guess_initiative_type(text: str) -> str:
+    text_without_diacritics = remove_diacritics(text)
+    for search_pattern, publication_type in INITIATIVE_TYPE_RULES:
+        if re.search(search_pattern, text_without_diacritics, re.IGNORECASE):
+            return publication_type
+    else:
+        return "OTHER"
+
+
+def cleanup_article_text(selector: Selector) -> str:
+    selector_copy = remove_html_tags(selector,
+                                     INITIATIVE_TEXT_CLEANUP_HTML_TAGS)
```

Comment:
TBH, nu sunt mare fan al valorilor extrase în constante, mi se pare că fac codul greu de urmărit. Aș prefera să văd `['ul', 'a']` în loc de `INITIATIVE_TEXT_CLEANUP_HTML_TAGS `. Same for all the others, cu excepția ălora mai stufoase, cum ar fi `ROMANIAN_MONTHS`.